src/Lean/Elab/Tactic/BvDecide/BitBlast/BVExpr/Basic.lean

```diff
@@ -0,0 +1,444 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Henrik Böving
+-/
+prelude
+import Init.Data.Hashable
+import Init.Data.BitVec
+import Lean.Elab.Tactic.BvDecide.BitBlast.BoolExpr.Basic
+
+/-!
+This module contains the definition of the `BitVec` fragment that `bv_decide` internally operates
+on as `BVLogicalExpr`. The preprocessing steps of `bv_decide` reduce all supported `BitVec`
+operations to the ones provided in this file. For verification purposes `BVLogicalExpr.Sat` and
+`BVLogicalExpr.Unsat` are provided.
+-/
+
+namespace Lean.Elab.Tactic.BvDecide
+
+/--
+The variable definition used by the bitblaster.
+-/
+structure BVBit where
+  /--
+  The width of the BitVec variable.
+  -/
+  {w : Nat}
+  /--
+  A numeric identifier for the BitVec variable.
+  -/
+  var : Nat
+  /--
+  The bit that we take out of the BitVec variable by getLsb.
+  -/
+  idx : Fin w
+  deriving Hashable, DecidableEq, Repr
+
+instance : ToString BVBit where
+  toString b := s!"x{b.var}[{b.idx.val}]"
+
+instance : Inhabited BVBit where
+  default := { w := 1, var := 0, idx := 0 }
+
+/--
+All supported binary operations on `BVExpr`.
+-/
+inductive BVBinOp where
+  /--
+  Bitwise and.
+  -/
+  | and
+  /--
+  Bitwise or.
+  -/
+  | or
+  /--
+  Bitwise xor.
+  -/
+  | xor
+  /--
+  Addition.
```

Comment:
Not sure what best practice is here, but I typically terminate `sentences` with `.` and expressions that lack a verb remain without terminator.